### PR TITLE
KAFKA-10619: Configure producer with idempotence and acks all by default (KIP-679)

### PR DESCRIPTION
**Pull request by warrenzhu25**
This PR relies on existing tests. A subsequent PR will make additional test adjustments
to ensure coverage of the non-default behavior is still good after this change.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by ctan888** [comment]
Hi, @warrenzhu25. We'll wait at least until 3.0 for merging as we are making the ACL changes. Please see KIP-679 for more details and join our discussion. Thanks.

**Comment by ijuma** [comment]
@d8tltanc can we rebase this PR so that we can get it merged?

**Comment by ijuma** [comment]
@d8tltanc I notice that the Jira ticket was assigned to you, but this PR was submitted by @warrenzhu25. Are you working together on this?

I am asking because we need to get this change into trunk soonish and changing the default is not the only change we require. We also need to check and potentially update the tests so that we get coverage of non idempotent cases too after the default is switched.

**Comment by ctan888** [comment]
@ijuma Could you help stamp this PR? I'll make the subsequent producer tests before the code freeze date. Thanks.

**Comment by ctan888** [comment]
Meanwhile, I'll check the failed tests.

**Comment by ijuma** [comment]
@d8tltanc Merged trunk and ran the tests here https://github.com/apache/kafka/pull/10950, failures are unrelated:

> Build / JDK 8 and Scala 2.12 / kafka.server.RaftClusterTest.testCreateClusterAndCreateAndManyTopics()
> Build / JDK 8 and Scala 2.12 / kafka.server.RaftClusterTest.testCreateClusterAndCreateListDeleteTopic() 
> Build / JDK 16 and Scala 2.13 / kafka.server.RaftClusterTest.testCreateClusterAndCreateAndManyTopicsWithManyPartitions()
> Build / JDK 15 and Scala 2.13 / org.apache.kafka.connect.mirror.integration.MirrorConnectorsIntegrationSSLTest.testReplication()
> Build / JDK 15 and Scala 2.13 / kafka.server.RaftClusterTest.testCreateClusterAndCreateAndManyTopicsWithManyPartitions()
> Build / JDK 15 and Scala 2.13 / kafka.server.RaftClusterTest.testCreateClusterAndCreateAndManyTopicsWithManyPartitions()

**Comment by ijuma** [comment]
@d8tltanc Can you please file a JIRA for the follow-up work?

1. Adjust config priority
2. Adjust the JUnit tests so we get good coverage of the non default behavior
3. Similar to 2 for system tests

File: clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java [hunk 1]
```diff
@@ -300,7 +300,7 @@
                                 .define(RETRIES_CONFIG, Type.INT, Integer.MAX_VALUE, between(0, Integer.MAX_VALUE), Importance.HIGH, RETRIES_DOC)
                                 .define(ACKS_CONFIG,
                                         Type.STRING,
-                                        "1",
+                                        "all",
                                         in("all", "-1", "0", "1"),
                                         Importance.HIGH,
```
Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java:305]: @d8tltanc in your PR with the test adjustments, please reduce the priority of this. Either `LOW` or `MEDIUM`, we can discuss more in the PR.

**Pull request merged**